search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -120,9 +120,22 @@ class TemporalEntityHandler(
             jsonLdEntityWithTemporalValues.contexts
         )
 
-        return ResponseEntity.status(HttpStatus.OK).body(serializeObject(filteredJsonLdEntity.compact()))
+        // WORKAROUND:
+        // the json-ld lib shows empty attributes as empty objects but we need empty arrays as stated in 5.7.4.4
+        // replace empty Maps by empty Lists
+        val postProcessedEntity = replaceEmptyMapsByEmptyLists(filteredJsonLdEntity.compact())
+
+        return ResponseEntity.status(HttpStatus.OK).body(serializeObject(postProcessedEntity))
     }
 
+    private fun replaceEmptyMapsByEmptyLists(compactedEntity: Map<String, Any>): Map<String, Any> {
```

Comment:
it could be moved to jsonLdUtils.

Reply:
If I move this function to jsonLdUtils the comment about the workaround should stay where it's called